src/features/tokens/CosmNativeTokenAdapter.ts

```diff
@@ -0,0 +1,135 @@
+import { MsgTransferEncodeObject } from '@cosmjs/stargate';
+import Long from 'long';
+
+import {
+  BaseAppAdapter,
+  ChainName,
+  CosmJsProvider,
+  CwHypCollateralAdapter,
+  ITokenAdapter,
+  MinimalTokenMetadata,
+  MultiProtocolProvider,
+  TransferParams,
+} from '@hyperlane-xyz/sdk';
+import { ProtocolType } from '@hyperlane-xyz/utils';
+
+const COSMOS_IBC_TRANSFER_TIMEOUT = 60_000; // 1 minute
+
+// TODO cosmos move everything in this file to the SDK
+export class BaseCosmosAdapter extends BaseAppAdapter {
+  public readonly protocol: ProtocolType = ProtocolType.Cosmos;
+
+  public getProvider(): CosmJsProvider['provider'] {
+    return this.multiProvider.getCosmJsProvider(this.chainName);
+  }
+}
+
+export class CosmNativeTokenAdapter extends BaseCosmosAdapter implements ITokenAdapter {
+  constructor(
+    public readonly chainName: ChainName,
+    public readonly multiProvider: MultiProtocolProvider,
+    public readonly addresses: Record<string, Address>,
+    public readonly properties: {
+      ibcDenom: string;
+      sourcePort: string;
+      sourceChannel: string;
+    },
+  ) {
+    if (!properties.ibcDenom || !properties.sourcePort || !properties.sourceChannel)
+      throw new Error('Missing properties for CosmNativeTokenAdapter');
+    super(chainName, multiProvider, addresses);
+  }
+
+  async getBalance(address: string): Promise<string> {
+    const provider = await this.getProvider();
+    const coin = await provider.getBalance(address, this.properties.ibcDenom);
+    return coin.amount;
+  }
+
+  getMetadata(): Promise<MinimalTokenMetadata> {
+    throw new Error('Metadata not available to native tokens');
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  populateApproveTx(_transferParams: TransferParams): unknown {
+    throw new Error('Approve not required for native tokens');
+  }
+
+  // TODO cosmos consider refactoring this into a new method b.c. the other adapter
```

Comment:
ahh yeah I was thinking this would be a `CosmosNativeToken implements IHypTokenAdapter `